internal/plugin/immutable_fields_test.go

```diff
@@ -0,0 +1,238 @@
+			assert, require := assert.New(t), require.New(t)
+			orig := newPlugin.testClonePlugin()
+			err := w.LookupById(context.Background(), orig)
+			require.NoError(err)
+
+			rowsUpdated, err := w.Update(context.Background(), tt.update, tt.fieldMask, nil, db.WithSkipVetForWrite(true))
+			require.Error(err)
+			assert.Equal(0, rowsUpdated)
+
+			after := newPlugin.testClonePlugin()
+			err = w.LookupById(context.Background(), after)
+			require.NoError(err)
+
+			assert.True(proto.Equal(orig, after))
+		})
+	}
+}
+
+func TestPluginVersion_ImmutableFields(t *testing.T) {
+	t.Parallel()
+	conn, _ := db.TestSetup(t, "postgres")
+	w := db.New(conn)
+
+	ts := timestamp.Timestamp{Timestamp: &timestamppb.Timestamp{Seconds: 0, Nanos: 0}}
+	plg := testPlugin(t, conn, "test")
+	plg2 := testPlugin(t, conn, "test2")
+	plgVer := testPluginVersion(t, conn, plg.GetPublicId(), "0.0.1")
+
+	newPluginVer := plgVer
+
+	tests := []struct {
+		name      string
+		update    *PluginVersion
+		fieldMask []string
+	}{
+		{
+			name: "public_id",
+			update: func() *PluginVersion {
+				c := newPluginVer.testClonePluginVersion()
+				c.PublicId = "hc_thisIsNotAValidId"
+				return c
+			}(),
+			fieldMask: []string{"PublicId"},
+		},
+		{
+			name: "create time",
+			update: func() *PluginVersion {
+				c := newPluginVer.testClonePluginVersion()
+				c.CreateTime = &ts
+				return c
+			}(),
+			fieldMask: []string{"CreateTime"},
+		},
+		{
+			name: "plugin_id",
+			update: func() *PluginVersion {
+				c := newPluginVer.testClonePluginVersion()
+				c.PluginId = plg2.GetPublicId()
+				return c
+			}(),
+			fieldMask: []string{"PluginId"},
+		},
+		{
+			name: "version",
+			update: func() *PluginVersion {
+				c := newPluginVer.testClonePluginVersion()
+				c.SemanticVersion = "0.0.2"
+				return c
+			}(),
+			fieldMask: []string{"SymanticVersion"},
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			assert, require := assert.New(t), require.New(t)
+			orig := newPluginVer.testClonePluginVersion()
+			err := w.LookupById(context.Background(), orig)
+			require.NoError(err)
+
+			rowsUpdated, err := w.Update(context.Background(), tt.update, tt.fieldMask, nil, db.WithSkipVetForWrite(true))
+			require.Error(err)
+			assert.Equal(0, rowsUpdated)
+
+			after := newPluginVer.testClonePluginVersion()
+			err = w.LookupById(context.Background(), after)
+			require.NoError(err)
+
+			assert.True(proto.Equal(orig, after))
+		})
+	}
+}
+
+func TestPluginExecutable_ImmutableFields(t *testing.T) {
+	t.Parallel()
+	conn, _ := db.TestSetup(t, "postgres")
+	w := db.New(conn)
+
+	ts := timestamp.Timestamp{Timestamp: &timestamppb.Timestamp{Seconds: 0, Nanos: 0}}
+	plg := testPlugin(t, conn, "test")
+	plgVer := testPluginVersion(t, conn, plg.GetPublicId(), "0.0.1")
+	plgExe := testPluginExecutable(t, conn, plgVer.GetPublicId(), "windows", "amd64", []byte("test"))
+
+	newPluginExe := plgExe
+
+	tests := []struct {
```

Comment:
Is the executable `bytea` column missing from the test set here? Just noticed there wasn't a case calling it out.